vmdb/spec/support/api_spec_helper.rb

```diff
@@ -116,9 +119,39 @@ def collection_config
     api_config[:collections]
   end
 
-  def action_identifier(type, action)
-    collection_config.fetch_path(type, :resource_actions, :post)
+  def action_identifier(type, action, selection = :resource_actions)
+    collection_config.fetch_path(type, selection, :post)
       .select { |spec| spec[:name] == action.to_s }
       .first[:identifier]
   end
+
+  def collection_action_identifier(type, action)
+    action_identifier(type, action, :collection_actions)
+  end
+
+  def subcollection_action_identifier(type, subtype, action)
+    action_identifier(type, action, "#{subtype}_subcollection_actions".to_sym)
+  end
+
+  def gen_request(action, *hrefs)
+    request = {"action" => action.to_s}
+    request["resources"] = hrefs.collect { |href| {"href" => href} } if hrefs.present?
+    request
+  end
+
+  def gen_requests(action, data)
+    request = {"action" => action.to_s}
+    request["resources"] = data
+    request
+  end
```

Comment:
This is identical to the previous method...could gen_request just call gen_requests?